server/odc-service/src/main/java/com/oceanbase/odc/service/common/DruidMonitor.java

```diff
@@ -0,0 +1,646 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.oceanbase.odc.service.common;
+
+import static com.oceanbase.odc.core.alarm.AlarmEventNames.DRUID_ACTIVE_COUNT_MORE_THAN_80_PERCENT;
+import static com.oceanbase.odc.core.alarm.AlarmEventNames.DRUID_MONITOR_ERROR;
+import static com.oceanbase.odc.core.alarm.AlarmEventNames.DRUID_WAIT_THREAD_COUNT_MORE_THAN_0;
+import static com.oceanbase.odc.core.alarm.AlarmEventNames.METHOD_TOO_LONG_EXECUTE_TIME;
+import static com.oceanbase.odc.core.alarm.AlarmEventNames.METHOD_TOO_MUCH_JDBC_EXECUTE_COUNT;
+import static com.oceanbase.odc.core.alarm.AlarmEventNames.SQL_EXECUTE_ERROR;
+import static com.oceanbase.odc.core.alarm.AlarmEventNames.SQL_TOO_LONG_EXECUTE_TIME;
+import static com.oceanbase.odc.core.alarm.AlarmEventNames.SQL_TOO_LONG_SQL_PARAMETERS;
+
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import com.alibaba.druid.pool.DruidDataSource;
+import com.alibaba.druid.stat.DruidStatManagerFacade;
+import com.alibaba.druid.support.spring.stat.SpringStatManager;
+import com.google.common.base.Preconditions;
+import com.oceanbase.odc.common.util.StringUtils;
+import com.oceanbase.odc.core.alarm.AlarmUtils;
+import com.oceanbase.odc.service.config.SystemConfigService;
+import com.oceanbase.odc.service.config.model.Configuration;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+@ConditionalOnProperty(value = "odc.monitor.enabled", havingValue = "true")
```

Comment:
may use key `odc.system.monitor.enabled` .
odc may offer user side datatabase monitor features in future.

Reply:
done